src/main/java/org/generationcp/middleware/service/impl/dataset/DatasetServiceImpl.java

```diff
@@ -1177,8 +1177,14 @@ private static boolean isDateOrCharacterDataType(final MeasurementVariable measu
 	void addStudyVariablesToUnitRows(final List<ObservationUnitRow> observationUnits, final List<MeasurementVariable> studyVariables) {
 		for (final ObservationUnitRow observationUnitRow : observationUnits) {
 			for (final MeasurementVariable measurementVariable : studyVariables) {
+				final String key;
+				if (measurementVariable.getAlias() != null && !measurementVariable.getAlias().equals("")) {
+					key = measurementVariable.getAlias();
+				} else {
+					key = measurementVariable.getName();
+				}
 				observationUnitRow.getVariables()
-					.put(measurementVariable.getName(), new ObservationUnitData(measurementVariable.getValue()));
+					.put(key, new ObservationUnitData(measurementVariable.getValue()));
```

Comment:
Is there any effect on saving/retrieving/export of data if the germplasm descriptor has an alias?

Reply:
Hi @abatac , 
Upon checking saving/export/retrieve result is the same as master.